model/delivery/container/LtiDeliveryContainer.php

```diff
@@ -24,29 +24,57 @@
 use oat\taoDelivery\model\container\execution\ExecutionClientContainer;
 use oat\taoDelivery\model\container\ExecutionContainer;
 use oat\taoDelivery\model\execution\DeliveryExecution;
+use IMSGlobal\LTI\ToolProvider\ToolConsumer;
+use oat\oatbox\session\SessionService;
+use oat\generis\model\OntologyAwareTrait;
+use oat\tao\model\oauth\DataStore;
 
 /**
  * Class LtiDeliveryContainer
  *
  * A delivery container to manage LTI based delivery
- *
- * @package oat\taoLtiConsumer\model\delivery\container
  */
 class LtiDeliveryContainer extends AbstractContainer
 {
+    use OntologyAwareTrait;
+
     /**
      * Get the execution container to render LTI based delivery
      *
      * @param DeliveryExecution $execution
+     *
      * @return ExecutionClientContainer|ExecutionContainer
-     * @throws \common_exception_NotImplemented
+     * @throws \common_exception_InvalidArgumentType
+     * @throws \common_exception_NotFound
      */
     public function getExecutionContainer(DeliveryExecution $execution)
     {
-        throw new \common_exception_NotImplemented('LTI delivery execution container is not yet implemented.');
-        $container = new ExecutionClientContainer($execution);
-        $container->setData('deliveryExecution', $execution->getIdentifier());
-        $container->setData('deliveryServerConfig', []);
+        $params = $this->getRuntimeParams();
```

Comment:
maybe we can split this fat method

Reply:
I'm not sure about splitting it. Better would be real Dependency Injection. Line 68 makes me cry/scream indeed...